src/theory/theory.cpp

```diff
@@ -515,16 +515,33 @@ bool Theory::areCareDisequal(TNode x, TNode y)
   Assert(d_equalityEngine != nullptr);
   Assert(d_equalityEngine->hasTerm(x));
   Assert(d_equalityEngine->hasTerm(y));
+  if (x.isConst() && y.isConst())
```

Comment:
```suggestion
  if (x.isConst() && y.isConst() && x != y)
```

Reply:
x != y can be an `Assert`.

Reply:
This actually needed to be added a separate case, done above now.